### PR TITLE
fix(radio): only emit change event with user input

### LINKED ISSUE #1523: Radio group change event should only fire with user interaction
Currently, if you set the `value` input programmatically, the change event fires. This behavior doesn't match Angular forms or native forms.

### PR DESCRIPTION
**Pull request by tinayuangao**
Closes #1523 

R: @kara @jelbourn

### CODE REVIEW

File: src/lib/radio/radio.spec.ts [hunk 1]
```diff
@@ -152,21 +152,22 @@ describe('MdRadio', () => {
       expect(spies[1]).toHaveBeenCalledTimes(1);
     });
 
-    it('should emit a change event from the radio group', () => {
+    it('should not emit a change event from the radio group when change group value '
+         + 'programmatically', () => {
```
Comment (kara) [inline src/lib/radio/radio.spec.ts]: Have you considered using backticks instead? That way you don't need the `+`

File: src/lib/radio/radio.spec.ts [hunk 2]
```diff
@@ -242,34 +243,36 @@ describe('MdRadio', () => {
 
       fixture.detectChanges();
 
-      expect(changeSpy).toHaveBeenCalled();
+      expect(changeSpy).toHaveBeenCalledTimes(0);
```
Comment (kara) [inline src/lib/radio/radio.spec.ts]: I think you can do 

```
expect(changeSpy).not.toHaveBeenCalled()
```

Might be more natural-sounding

File: src/lib/radio/radio.ts [hunk 3]
```diff
@@ -171,6 +166,13 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
     }
   }
 
+  emitChangeEvent(): void {
```
Comment (kara) [inline src/lib/radio/radio.ts]: Did you want this to be part of the public API?  It seems to me like it should use the `_` prefix convention so people know it's internal (only used by the button)

**Comment by tinayuangao** [comment]
Comment addressed. Please take another look. Thanks for review!

**Pull request merged**